File: src/main/webapp/na-spolke-client/src/classes/financialStatementProtocol/ResolutionApprovalBodyMember.js
import {Resolution} from "./Resolution";

export class ResolutionApprovalBodyMember extends Resolution{
    bodyType;
    boardMember;
    director;
    wholeReportingPeriod;
    beginningOfReportingPeriod;
    endOfReportingPeriod;


    constructor(data,votingType, resolutionTitle, key, member) {
        super(data, votingType, resolutionTitle, key, member[key==="board" ? "boardMemberId" : "boardOfDirectorId"]);
        this.bodyType = key;
        if(this.bodyType==="board"){
            this.boardMember = member;
        } else {
            this.director = member;
        }
        this.wholeReportingPeriod = data[`${key}${member[key==="board" ? "boardMemberId" : "boardOfDirectorId"]}WholeReportingPeriod`]
        if(this.wholeReportingPeriod===false) {
            this.beginningOfReportingPeriod = data[`${key}${member[key==="board" ? "boardMemberId" : "boardOfDirectorId"]}Beginning`]
            this.endOfReportingPeriod = data[`${key}${member[key==="board" ? "boardMemberId" : "boardOfDirectorId"]}End`]
        }
    }
}

//[bodyType==="board" ? "boardMemberId" : "boardOfDirectorId"]

File: src/main/java/com/company/naspolke/model/company/financialStatements/resolutions/ResolutionApprovalBodyMember.java
package com.company.naspolke.model.company.financialStatements.resolutions;

import com.company.naspolke.model.company.companyBodies.BoardMember;
import com.company.naspolke.model.company.companyBodies.BoardOfDirector;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class ResolutionApprovalBodyMember implements VotingInterface {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;
    private String votingType;
    private String resolutionTitle;
    private boolean unanimously;
    private int votesFor;
    private int votesAgainst;
    private int votesAbstentions;
    private String bodyType;
    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name = "board_member_board_member_id")
    private BoardMember boardMember;
    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name = "director_board_of_director_id")
    private BoardOfDirector director;
    private boolean wholeReportingPeriod;
    private LocalDate beginningOfReportingPeriod;
    private LocalDate endOfReportingPeriod;
}
